.ci/ci-fast-return.sh

```diff
@@ -598,6 +409,17 @@ testCheckSkipLabel() {
 	assertEquals "1" "$result"
 }
 
+
+# There functions moved to lib.sh, but we still want to test them here, so call
+# the test method there
+testIsLabelSet() {
+	lib_testIsLabelSet
+}
+
+testCheckLabel() {
+	lib_testCheckLabel
+}
```

Comment:
Can't we just `shunit2` `lib.sh`?

Reply:
Hey @Jakob-Naucke - I might have misunderstood you, but I originally tried adding `"${GOPATH}/src/${shunit2_path}/shunit2" lib.sh` to the `self_test`

and I got:
```
shunit2:FATAL unable to read from lib.sh
```
I'm guessing I did something wrong there?

Reply:
I meant: Can't we just make `lib.sh` another file that has self-tests, and not test these functions from `ci-fast-return.sh`/`resolve-kata-dependencies.sh`?

Reply:
Sorry for misunderstanding. 

The reason I stayed away from that approach was that I wasn't sure if we wanted a `main` function (which would execute the shunit tests if the `test` argument was applied) added to `lib.sh` as it is a common library script. It has a bunch of code that runs outside of its functions and is sourced in a lot of places, so we'd end up executing main a lot of times (which would do nothing if it didn't have parameters).

I'm happy to give it a go though if you think it is worth trying.

Reply:
No need to apologise, I was too brief :)
DYT something like [`[ "$0" != "$BASH_SOURCE" ]`](https://stackoverflow.com/a/23009039) would work for introducing a main function? This slightly resembles Python's `if __name__ == "__main__"`, which is only true when the file itself is executed, but not when imported.

Reply:
> DYT something like [`[ "$0" != "$BASH_SOURCE" ]`](https://stackoverflow.com/a/23009039) would work for introducing a main function? This slightly resembles Python's `if __name__ == "__main__"`, which is only true when the file itself is executed, but not when imported.

I really like that idea. I've had an attempt at it in https://github.com/kata-containers/tests/pull/4363/commits/a48e644b740ff862146279f6aa6bb1cc846b1c10. If you are happy with it and all the test look okay then I'll squash it into the unit tests commit, but I've kept it separate for now whilst this bit is 'in progress'